app/src/main/java/com/epam/caloriecalc/ui/addnew/AddItemViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.epam.caloriecalc.ui.addnew
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.epam.caloriecalc.data.local.entities.IntakeRecord
+import com.epam.caloriecalc.data.local.repository.CalorieRepository
+import com.epam.caloriecalc.util.AddItemEvent
+import com.epam.caloriecalc.util.SnackbarActionType
+import com.epam.caloriecalc.util.UiEvent
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AddItemViewModel @Inject constructor(
+    internal val repository: CalorieRepository
+) : ViewModel() {
+
+    init {
+        viewModelScope.launch {
+            for (item in testitems)
+                repository.insertProduct(item)
+        }
+    }
+
+    private lateinit var pendingItem: IntakeRecord
+    private val _uiEvent = Channel<UiEvent>()
+    val uiEvent = _uiEvent.receiveAsFlow()
+
+    fun onEvent(event: AddItemEvent) {
+        when (event) {
+            is AddItemEvent.OnAddItemClick -> {
+                viewModelScope.launch {
+                    repository.insertIntake(IntakeRecord(productId = event.product.productId))
```

Comment:
I think with Room you could also just return an id from an insert operation - [Docs link](https://developer.android.com/training/data-storage/room/accessing-data#convenience-insert)
So then you could replace that `getLastIntake()` method with a `deleteByIntakeId` query, for which you could set and hold the intakeId here, so it can be used for the snackbar undo, then that would not have to query the database when pressed